lib/matplotlib/backend_bases.py

```diff
@@ -2040,12 +2044,68 @@ def draw_idle(self, *args, **kwargs):
             with self._idle_draw_cntx():
                 self.draw(*args, **kwargs)
 
+    @property
+    def device_pixel_ratio(self):
+        """
+        The ratio of logical to physical pixels used for the canvas on screen.
+
+        By default, this is 1, meaning physical and logical pixels are the same
+        size. Subclasses that support High DPI screens may set this property to
+        indicate that said ratio is different. All Matplotlib interaction,
+        unless working directly with the canvas, remains in logical pixels.
+
+        """
+        return self._device_pixel_ratio
+
+    def _set_device_pixel_ratio(self, ratio):
+        """
```

Comment:
Worth noting in the docstring that this will also also automatically update the figure DPI?

Reply:
Added.